jujugui/static/gui/src/app/components/deployment-flow/deployment-flow.js

```diff
@@ -69,21 +68,42 @@ YUI.add('deployment-flow', function() {
         credential: this.props.credential,
         loggedIn: !!this.props.getAuth(),
         modelName: this.props.modelName,
+        newTerms: [],
         region: this.props.region,
         showChangelogs: false,
         sshKey: null,
-        terms: this._getTerms(),
+        terms: this._getTerms() || [],
+        termsList: [],
```

Comment:
We've got three different terms-ish items in state; maybe in-line comments on each to provide a little clarification on how they differ?